File: go-sdk/pkg/models/operations/listdatabases.go
package operations

type ListDatabasesPathParams struct {
	Organization string `pathParam:"style=simple,explode=false,name=organization"`
}

type ListDatabasesQueryParams struct {
	Page    *float64 `queryParam:"style=form,explode=true,name=page"`
	PerPage *float64 `queryParam:"style=form,explode=true,name=per_page"`
}

type ListDatabasesRequest struct {
	PathParams  ListDatabasesPathParams
	QueryParams ListDatabasesQueryParams
}

type ListDatabasesResponse struct {
	ContentType                           string
	StatusCode                            int64
	ListDatabases200ApplicationJSONObject map[string]map[string]interface{}
}

File: ts-sdk/src/sdk/models/operations/listdatabases.ts
import { SpeakeasyBase, SpeakeasyMetadata } from "../../../internal/utils";
import { Type } from "class-transformer";


export class ListDatabasesPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=organization" })
  organization: string;
}

export class ListDatabasesQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=page" })
  page?: number;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=per_page" })
  perPage?: number;
}

export class ListDatabasesRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: ListDatabasesPathParams;

  @SpeakeasyMetadata()
  queryParams: ListDatabasesQueryParams;
}

export class ListDatabasesResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  listDatabases200ApplicationJSONObject?: Record<string, Record<string, any>>;
}